cf-agent/package_module.c

```
@@ -0,0 +1,1611 @@
+{
+    Bundle bundle = {.name = "package_global"};
+    PromiseType promise_type = {.name = "package_global",
+                               .parent_bundle = &bundle};
+    Promise pp = {.promiser = "package_global",
+                  .parent_promise_type = &promise_type};
+    
+    CfLock package_promise_global_lock;
+    
+    package_promise_global_lock =
+            AcquireLock(ctx, GLOBAL_PACKAGE_PROMISE_LOCK_NAME, VUQNAME, CFSTARTTIME,
+                        (TransactionContext) {.ifelapsed = VIFELAPSED,
+                                              .expireafter = VEXPIREAFTER},
+                        &pp, false);
+                        
+    return (PackagePromiseGlobalLock) {.g_lock = package_promise_global_lock,
+                                       .lock_ctx = ctx};
+}
+
+void YieldGlobalPackagePromiseLock(PackagePromiseGlobalLock lock)
+{
+    Bundle bundle = {.name = "package_global"};
+    PromiseType promise_type = {.name = "package_global",
+                               .parent_bundle = &bundle};
+    Promise pp = {.promiser = "package_global",
+                  .parent_promise_type = &promise_type};
+    
+    YieldCurrentLockAndRemoveFromCache(lock.lock_ctx, lock.g_lock,
+                                       GLOBAL_PACKAGE_PROMISE_LOCK_NAME, &pp);
+}
+
+static void ParseAndLogErrorMessage(const Rlist *data)
+{
+    for (const Rlist *rp = data; rp != NULL; rp = rp->next)
+    {
+        char *line = RlistScalarValue(rp);
+                   
+        if (StringStartsWith(line, "Error="))
+        {
+            Log(LOG_LEVEL_VERBOSE, "Caught error: %s", line + strlen("Error="));
+        }
+        else if (StringStartsWith(line, "ErrorMessage="))
+        {
+            Log(LOG_LEVEL_VERBOSE, "Caught error message: %s", 
+                line + strlen("ErrorMessage="));
+        }
+        else
+        {
+            Log(LOG_LEVEL_VERBOSE, "Caught unsupported error info: %s", line);
+        }
```

Makes sense. At the moment ignoring error messages (no error recovery) and only action performed is logging.